---
title: "Spotify"
author: "Sophia Stang"
date: 2025-03-03
categories: [python, pandas, spotify] # tags for a blog post (e.g., python)
image: "Britney_Spears.jpg"

toc: true
---

# Part 4. Jupyter Notebook Blogging
Write a blog post about your favorite artist(s) in the spotify DataFrame using Jupyter Notebook, and add it to your online blog.
In your blog post, utilize counting, sorting, indexing, and filtering methods.


In [ ]:
import pandas as pd

In [ ]:
spotify = pd.read_csv('https://bcdanl.github.io/data/spotify_all.csv')
spotify

pid playlist_name  pos        artist_name  \
0            0    Throwbacks    0      Missy Elliott   
1            0    Throwbacks    1     Britney Spears   
2            0    Throwbacks    2            Beyoncé   
3            0    Throwbacks    3  Justin Timberlake   
4            0    Throwbacks    4             Shaggy   
...        ...           ...  ...                ...   
198000  999998            ✝️    6       Chris Tomlin   
198001  999998            ✝️    7       Chris Tomlin   
198002  999998            ✝️    8            Crowder   
198003  999998            ✝️    9       Chris Tomlin   
198004  999998            ✝️   10   Caleb and Kelsey   

                                        track_name  duration_ms  \
0       Lose Control (feat. Ciara & Fat Man Scoop)       226863   
1                                            Toxic       198800   
2                                    Crazy In Love       235933   
3                                   Rock Your Body       267266   
4                                     It Wasn't Me       227600   
...                                            ...          ...   
198000                                   Waterfall       209573   
198001                                    The Roar       220106   
198002        Lift Your Head Weary Sinner (Chains)       224666   
198003                                We Fall Down       280960   
198004      10,000 Reasons / What a Beautiful Name       178189   

                                            album_name  
0                                         The Cookbook  
1                                          In The Zone  
2         Dangerously In Love (Alben für die Ewigkeit)  
3                                            Justified  
4                                             Hot Shot  
...                                                ...  
198000                                    Love Ran Red  
198001                                    Love Ran Red  
198002                                    Neon Steeple  
198003  How Great Is Our God: The Essential Collection  
198004          10,000 Reasons / What a Beautiful Name  

[198005 rows x 7 columns]

I really enjoy the spotify playlist "Throwbacks".

In [ ]:
throwbacks = spotify['playlist_name'] == "Throwbacks"
throwbacks_df = spotify[throwbacks]
throwbacks_df

pid playlist_name  pos        artist_name  \
0            0    Throwbacks    0      Missy Elliott   
1            0    Throwbacks    1     Britney Spears   
2            0    Throwbacks    2            Beyoncé   
3            0    Throwbacks    3  Justin Timberlake   
4            0    Throwbacks    4             Shaggy   
...        ...           ...  ...                ...   
186699  999823    Throwbacks    5      Nelly Furtado   
186700  999823    Throwbacks    6              Usher   
186701  999823    Throwbacks    7               Iyaz   
186702  999823    Throwbacks    8           Jay Sean   
186703  999823    Throwbacks    9  Far East Movement   

                                        track_name  duration_ms  \
0       Lose Control (feat. Ciara & Fat Man Scoop)       226863   
1                                            Toxic       198800   
2                                    Crazy In Love       235933   
3                                   Rock Your Body       267266   
4                                     It Wasn't Me       227600   
...                                            ...          ...   
186699                                 Promiscuous       242293   
186700                   DJ Got Us Fallin' In Love       220800   
186701                                      Replay       182306   
186702                                        Down       212506   
186703                                   Like A G6       216893   

                                          album_name  
0                                       The Cookbook  
1                                        In The Zone  
2       Dangerously In Love (Alben für die Ewigkeit)  
3                                          Justified  
4                                           Hot Shot  
...                                              ...  
186699                                         Loose  
186700            Raymond v Raymond (Deluxe Edition)  
186701                                        Replay  
186702                                All Or Nothing  
186703                                    Free Wired  

[627 rows x 7 columns]

In [ ]:
throwbacks_df['artist_name'].nunique()

223

There is a total of 223 different artists in this playlist, including one of my favorite artists, Britney Spears.

In [ ]:
by_artist = throwbacks_df.set_index(keys = "artist_name")
by_artist

pid playlist_name  pos  \
artist_name                                    
Missy Elliott           0    Throwbacks    0   
Britney Spears          0    Throwbacks    1   
Beyoncé                 0    Throwbacks    2   
Justin Timberlake       0    Throwbacks    3   
Shaggy                  0    Throwbacks    4   
...                   ...           ...  ...   
Nelly Furtado      999823    Throwbacks    5   
Usher              999823    Throwbacks    6   
Iyaz               999823    Throwbacks    7   
Jay Sean           999823    Throwbacks    8   
Far East Movement  999823    Throwbacks    9   

                                                   track_name  duration_ms  \
artist_name                                                                  
Missy Elliott      Lose Control (feat. Ciara & Fat Man Scoop)       226863   
Britney Spears                                          Toxic       198800   
Beyoncé                                         Crazy In Love       235933   
Justin Timberlake                              Rock Your Body       267266   
Shaggy                                           It Wasn't Me       227600   
...                                                       ...          ...   
Nelly Furtado                                     Promiscuous       242293   
Usher                               DJ Got Us Fallin' In Love       220800   
Iyaz                                                   Replay       182306   
Jay Sean                                                 Down       212506   
Far East Movement                                   Like A G6       216893   

                                                     album_name  
artist_name                                                      
Missy Elliott                                      The Cookbook  
Britney Spears                                      In The Zone  
Beyoncé            Dangerously In Love (Alben für die Ewigkeit)  
Justin Timberlake                                     Justified  
Shaggy                                                 Hot Shot  
...                                                         ...  
Nelly Furtado                                             Loose  
Usher                        Raymond v Raymond (Deluxe Edition)  
Iyaz                                                     Replay  
Jay Sean                                         All Or Nothing  
Far East Movement                                    Free Wired  

[627 rows x 6 columns]

In [ ]:
britney = throwbacks_df['artist_name'] == "Britney Spears"
britney_df = throwbacks_df[britney]
britney_df

pid playlist_name  pos     artist_name  \
1            0    Throwbacks    1  Britney Spears   
24216      380    Throwbacks    1  Britney Spears   
24245      380    Throwbacks   30  Britney Spears   
24340      380    Throwbacks  125  Britney Spears   
48261      717    Throwbacks   51  Britney Spears   
48262      717    Throwbacks   52  Britney Spears   
60323      895    Throwbacks  121  Britney Spears   
60324      895    Throwbacks  122  Britney Spears   
83553     1247    Throwbacks    3  Britney Spears   
83554     1247    Throwbacks    4  Britney Spears   
83555     1247    Throwbacks    5  Britney Spears   
83556     1247    Throwbacks    6  Britney Spears   
83615     1247    Throwbacks   65  Britney Spears   
83644     1247    Throwbacks   94  Britney Spears   
83645     1247    Throwbacks   95  Britney Spears   
135830  999022    Throwbacks   64  Britney Spears   

                             track_name  duration_ms  \
1                                 Toxic       198800   
24216                            Circus       192360   
24245                                 3       213173   
24340                         Womanizer       224400   
48261                             Toxic       198800   
48262             ...Baby One More Time       211066   
60323                          Stronger       203000   
60324                             Lucky       206226   
83553            Oops!...I Did It Again       211160   
83554                             Toxic       198800   
83555                   I'm a Slave 4 U       203600   
83556                       Piece of Me       212106   
83615                             Lucky       206226   
83644   I'm Not a Girl, Not Yet a Woman       231066   
83645                        Gimme More       251240   
135830                           Circus       192360   

                                            album_name  
1                                          In The Zone  
24216                          Circus (Deluxe Version)  
24245                           The Singles Collection  
24340                          Circus (Deluxe Version)  
48261                                      In The Zone  
48262   ...Baby One More Time (Digital Deluxe Version)  
60323                          Oops!... I Did It Again  
60324                          Oops!... I Did It Again  
83553                          Oops!... I Did It Again  
83554                                      In The Zone  
83555                 Britney (Digital Deluxe Version)  
83556                                         Blackout  
83615                          Oops!... I Did It Again  
83644                 Britney (Digital Deluxe Version)  
83645                                         Blackout  
135830                         Circus (Deluxe Version)

In [ ]:
britney_df['track_name'].count()

16

She has 16 different songs in the playlist:

In [ ]:
britney_df['track_name']

1                                   Toxic
24216                              Circus
24245                                   3
24340                           Womanizer
48261                               Toxic
48262               ...Baby One More Time
60323                            Stronger
60324                               Lucky
83553              Oops!...I Did It Again
83554                               Toxic
83555                     I'm a Slave 4 U
83556                         Piece of Me
83615                               Lucky
83644     I'm Not a Girl, Not Yet a Woman
83645                          Gimme More
135830                             Circus
Name: track_name, dtype: object

In [ ]:
song_length = britney_df['duration_ms'].sort_values(ascending=False)
song_length
song_length_min = song_length/60000
song_length_min

83645     4.187333
83644     3.851100
24340     3.740000
24245     3.552883
83556     3.535100
83553     3.519333
48262     3.517767
60324     3.437100
83615     3.437100
83555     3.393333
60323     3.383333
1         3.313333
48261     3.313333
83554     3.313333
24216     3.206000
135830    3.206000
Name: duration_ms, dtype: float64

Her longest song is 4.187 minutes, while her shortest song is 3.206 minutes.

In [ ]:
song_length_min.sum()

55.90638333333333

She makes up a total of 55.906 minutes of the playlist "Throwbacks".